src/client/api/restful/get_outputs_id.c

```diff
@@ -6,6 +6,172 @@
+        break;
+      case QUERY_PARAM_GOV:
+        query_str_len += strlen(OUTPUTS_QUERY_GOV_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_ISSUER:
+        query_str_len += strlen(OUTPUTS_QUERY_ISSUER_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      default:
+        break;
+    }
+  }
+  query_str_len--;  // Remove the "&" params seperator at the end
+  return query_str_len;
+}
+
+static int copy_param_to_buf(char *buf, size_t offset, char *key, outputs_query_list_t *elm) {
+  int len = offset;
+  memcpy(buf + offset, key, strlen(key));
+  offset += strlen(key);
+  buf[offset++] = '=';
+  memcpy(buf + offset, elm->query_item->param, strlen(elm->query_item->param));
+  offset += strlen(elm->query_item->param);
+  buf[offset++] = '&';
+  return offset - len;
+}
+
+size_t get_outputs_query_str(outputs_query_list_t *list, char *buf, size_t buf_len) {
+  // Check if buffer length is sufficient for holding query string
+  size_t query_str_len = get_outputs_query_str_len(list);
+  if (buf_len < query_str_len + 1) {
+    printf("[%s:%d] buffer length not sufficient\n", __func__, __LINE__);
+  }
+
+  size_t offset = 0;
+  outputs_query_list_t *elm;
+  LL_FOREACH(list, elm) {
+    switch (elm->query_item->type) {
+      case QUERY_PARAM_ADDRESS:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_ADDRESS_KEY, elm);
+        break;
+      case QUERY_PARAM_DUST_RET:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_DUST_RET_KEY, elm);
+        break;
+      case QUERY_PARAM_DUST_RET_ADDR:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_DUST_RET_ADDR_KEY, elm);
+        break;
+      case QUERY_PARAM_SENDER:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_SENDER_KEY, elm);
+        break;
+      case QUERY_PARAM_TAG:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_TAG_KEY, elm);
+        break;
+      case QUERY_PARAM_PAGE_SIZE:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_PAGE_SIZE_KEY, elm);
+        break;
+      case QUERY_PARAM_CURSOR:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_CURSOR_KEY, elm);
+        break;
+      case QUERY_PARAM_STATE_CTRL:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_STATE_CTRL_KEY, elm);
+        break;
+      case QUERY_PARAM_GOV:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_GOV_KEY, elm);
+        break;
+      case QUERY_PARAM_ISSUER:
+        offset += copy_param_to_buf(buf, offset, OUTPUTS_QUERY_ISSUER_KEY, elm);
+        break;
+      default:
+        break;
+    }
+  }
+  buf[offset - 1] = 0;  // Replace the "&" at the end with '\0'
```

Comment:
What if query list is empty?

Reply:
Added check for query_str_len equals zero at the begining.
Also added check for buf[offset-1] == '&' at the end. If list is empty it returns 0.